x/gamm/pool-models/balancer/pool_test.go

```diff
@@ -469,6 +469,51 @@ func TestGetPoolAssetsByDenom(t *testing.T) {
 	}
 }
 
+func (suite *KeeperTestSuite) TestEnsureDenomInPool() {
+	tests := map[string]struct {
+		poolAssets  []balancer.PoolAsset
+		tokensIn    sdk.Coins
+		expectPass  bool
+		expectedErr error
+	}{
+		"all of tokensIn is in pool asset map": {
+			poolAssets:  []balancer.PoolAsset{defaultOsmoPoolAsset, defaultAtomPoolAsset},
+			tokensIn:    sdk.NewCoins(sdk.NewCoin("uatom", sdk.OneInt())),
+			expectPass:  true,
+			expectedErr: nil,
+		},
+		"one of tokensIn is in pool asset map": {
+			poolAssets:  []balancer.PoolAsset{defaultOsmoPoolAsset, defaultAtomPoolAsset},
+			tokensIn:    sdk.NewCoins(sdk.NewCoin("uatom", sdk.OneInt()), sdk.NewCoin("foo", sdk.OneInt())),
+			expectPass:  false,
+			expectedErr: types.ErrDenomNotFoundInPool,
+		},
+		"none of tokensIn is in pool asset map": {
```

Comment:
Please consider a case where `poolAssets` are empty. I know it shouldn't be possible considering the system in macro. However, it would still be great to test this function with any possible combination of inputs

Reply:
Added!